Truncate branch name when used in version

Dependabot creates insanely long branch names, resulting in the following build error in PRs:

> error NU5123: Warning As Error: The file 'package/services/metadata/core-properties/2edd8f9e5f1f4ccf939146ba2603f884.psmdcp' path, name, or both are too long. Your package might not work without long file path support. Please shorten the file path or file name.

We should truncate long branch names to make the PR builds from Dependabot succeed.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -85,8 +85,8 @@ jobs:
             # Get the version suffix from the auto-incrementing build number. For example: '123' => 'master-00123'
             $revision = "{0:D5}" -f [convert]::ToInt32($env:GITHUB_RUN_NUMBER, 10)
             $branchName = ![string]::IsNullOrEmpty($env:GITHUB_HEAD_REF) ? $env:GITHUB_HEAD_REF : $env:GITHUB_REF_NAME
-            $safeName = $branchName.Replace('/', '-').Replace('_', '-')
-            $versionSuffix = "$safeName-$revision"
+            $safeBranchName = $($branchName -Replace '[^a-zA-Z0-9-]', '-')[0..40] -Join ""
+            $versionSuffix = "$safeBranchName-$revision"
         }
         Write-Output "Using version suffix: $versionSuffix"
         Write-Output "PACKAGE_VERSION_SUFFIX=$versionSuffix" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append

